src/diff/diffDirectories.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import path from "path";
+import _ from "lodash";
+import { diffRaml, RamlDiff } from "./diffProcessor";
+import { ramlToolLogger } from "../common/logger";
+
+/**
+ * Extracts all the RAML file names from the specified config file.
+ *
+ * @param configPath - Target config file
+ *
+ * @returns An array of `assetId/fileName` for every api in the input file
+ */
+export function listRamlsFromConfig(configPath: string): string[] {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const config = require(configPath);
+  const ramls = [];
+  for (const apiFamily of Object.keys(config)) {
+    for (const api of config[apiFamily]) {
+      ramls.push(path.join(api.assetId, api.fatRaml.mainFile));
+    }
+  }
+
+  return ramls;
+}
+
+/**
+ * Compares two arrays and returns their intersection and symmetric difference.
+ *
+ * @param leftArr - One of the arrays being compared
+ * @param rightArr - The other array being compared
+ *
+ * @returns An array each for all the common, exclusive to left and exclusive to
+ * right elements.
+ */
+function compareArrays<T>(
+  leftArr: T[],
+  rightArr: T[]
+): {
+  leftOnly: T[];
+  rightOnly: T[];
+  common: T[];
+} {
+  const left = new Set(leftArr);
```

Comment:
Do we want to eliminate duplicate entries? If left contains [1,1,3] and right contains [1,3] this would basically show everything to be in common when there is an extra 1 on the left side.

Reply:
The function is only ever called with the results of `listRamlsFromConfig`. Will that function ever return an array with duplicates?

https://github.com/SalesforceCommerceCloud/raml-toolkit/blob/70b0a165d72606b9ccb4ab2eba00e0a8441bda5e/src/diff/diffDirectories.ts#L26

Reply:
Then we should more specifically name this method and describe what it does.  IT was written to seem like a generic array differ but it isn't that so we should be explicit.